doc/migrations/v0.29-v0.30.md

```diff
@@ -0,0 +1,191 @@
+This default value was changed on the pubsub router layer in the past, but we kept it overwritten in libp2p to avoid an upstream breaking change.
+Now `js-libp2p` does not overwrite the pubsub router options anymore. Upstream projects that want this feature should enable it on their libp2p configuration.
+
+**Before**
+
+```js
+const Gossipsub = require('libp2p-gossipsub')
+const Libp2p = require('libp2p')
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+    pubsub: Gossipsub
+  }
+})
+```
+
+**After**
+
+```js
+const Gossipsub = require('libp2p-gossipsub')
+const Libp2p = require('libp2p')
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+    pubsub: Gossipsub
+  },
+  config: {
+    pubsub: {
+      emitSelf: true
+    }
+  }
+})
+```
+
+The [Pubsub interface](https://github.com/libp2p/js-libp2p-interfaces/tree/master/src/pubsub) was updated on its message signing properties, taking into account the Gossipsub spec updates on [libp2p/specs#294](https://github.com/libp2p/specs/pull/294) and [libp2p/specs#299](https://github.com/libp2p/specs/pull/299)
+
+The signing property is now based on a `globalSignaturePolicy` option instead of the previous `signMessages` and `strictSigning` options. The default to strict signing pubsub messages was kept, but if you would like to disable it, the properties should be changed as follows:
+
+**Before**
+
+```js
+const Gossipsub = require('libp2p-gossipsub')
+const Libp2p = require('libp2p')
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+    pubsub: Gossipsub
+  },
+  config: {
+    pubsub: {
+      signMessages: false,
+      strictSigning: false
+    }
+  }
+})
+```
+
+**After**
+
+```js
+const Gossipsub = require('libp2p-gossipsub')
+const { SignaturePolicy } = require('libp2p-interfaces/src/pubsub/signature-policy')
+const Libp2p = require('libp2p')
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+    pubsub: Gossipsub
+  },
+  config: {
+    pubsub: {
+      globalSignaturePolicy: SignaturePolicy.StrictNoSign
+    }
+  }
+})
+```
+
+### Addresses
+
+Libp2p has supported `noAnnounce` addresses configuration for some time now. However, it did not provide the best developer experience. In this release, we dropped the `noAnnounce` configuration property in favor of an `announceFilter` property function.
+
+**Before**
+
+```js
+const Libp2p = require('libp2p')
+
+// ... Create/Load PeerId
+
+const libp2p = await Libp2p.create({
+  modules: {
+    // ... Add required modules according to the Configuration docs
+  },
+  peerId,
+  addresses: {
+    listen: ['/ip4/127.0.0.1/tcp/8000/ws'],
+    noAnnounce: ['/ip4/127.0.0.1/tcp/8000/ws'],
+  }
+})
```

Comment:
`peerID` isn't necessary in here, nor is modules. I would just truncate for simplicity:
```suggestion
const libp2p = await Libp2p.create({
  addresses: {
    listen: ['/ip4/127.0.0.1/tcp/8000/ws'],
    noAnnounce: ['/ip4/127.0.0.1/tcp/8000/ws'],
  },
  // ... additional configuration per the Configuration docs
})
```